rapid7_insightidr/komand_rapid7_insightidr/actions/close_investigations_in_bulk/action.py

```diff
@@ -0,0 +1,72 @@
+import komand
+from .schema import CloseInvestigationsInBulkInput, CloseInvestigationsInBulkOutput, Input, Output, Component
+from komand.exceptions import PluginException
+# Custom imports below
+from komand_rapid7_insightidr.util.endpoints import Investigations
+from komand_rapid7_insightidr.util.resource_helper import ResourceHelper
+import json
+from datetime import datetime
+from datetime import timedelta
+
+
+class CloseInvestigationsInBulk(komand.Action):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name='close_investigations_in_bulk',
+            description=Component.DESCRIPTION,
+            input=CloseInvestigationsInBulkInput(),
+            output=CloseInvestigationsInBulkOutput())
+
+    def run(self, params={}):
+        request = ResourceHelper(self.connection.session, self.logger)
+        endpoint = Investigations.close_investigations_in_bulk(self.connection.url)
+
+        source = self._get_with_default(params, Input.SOURCE, "MANUAL")
+        max_investigations_to_close = self._get_with_default(params, Input.MAX_INVESTIGATIONS_TO_CLOSE, None)
+        alert_type = self._get_with_default(params, Input.ALERT_TYPE, None)
```

Comment:
I might be missing something here, but why can't we do params.get(Input.SOURCE, "MANUAL") instead of using the function?

Reply:
Because `get` return default only when value in this key is `None`. If value is empty string, then `get` will return empty string instead "MANUAL". This input is not required so when user in WebUI choose this action without type `source` input then WebUI send to action empty string in this field. So we can't use method `get` with inputs that are not required.

Reply:
Ahhh I see now, thanks for the clarification!